ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -136,32 +145,134 @@ const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
-            'label': institution.displayName ? institution.displayName : institution.shortName,
-            'value': {displayName: institution.displayName, shortName: institution.shortName}
-          };
-        },
-        institutionsResponse.institutions
-      );
-      this.setState({verifiedInstitutionOptions: options});
+      const institutionsResponse = await institutionApi().getPublicInstitutionDetails();
+      const institutions = institutionsResponse.institutions;
+      this.setState({
+        verifiedInstitutionOptions: fp.sortBy( institution => institution.displayName, institutions),
+        verifiedInstitutionsByShortname: institutions.reduce(
+          (accumulator, institution) => accumulator.set(institution.shortName, institution),
+          new Map<String, PublicInstitutionDetails>()
+        )
+      });
     } catch (error) {
       this.setState({institutionsLoadingError: 'Could not get list of verified institutions - please try again later'});
     }
   }
 
+  getRoleOptionsForProfile() {
+    const {updatedProfile: {verifiedInstitutionalAffiliation}, verifiedInstitutionOptions} = this.state;
+    const institutionShortName = verifiedInstitutionalAffiliation ? verifiedInstitutionalAffiliation.institutionShortName : '';
+    return getRoleOptions(verifiedInstitutionOptions, institutionShortName);
+  }
+
+  getInstitutionDropdownOptions() {
+    const {verifiedInstitutionOptions} = this.state;
+    return fp.map(
+      institution => {
+        return {label: institution.displayName, value: institution.shortName};
+      },
+      verifiedInstitutionOptions
+    );
+  }
+
+  isSaveDisabled(errors) {
+    const {oldProfile, updatedProfile} = this.state;
+    return fp.isEqual(oldProfile, updatedProfile) || errors;
+  }
+
+  setVerifiedInstitutionOnProfile(institutionShortName: string) {
+    const {verifiedInstitutionsByShortname} = this.state;
+    this.setState(
+      fp.set(
+          ['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionShortName'],
+        institutionShortName
+      )
+    );
+    this.setState(
+      fp.set(
+          ['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionDisplayName'],
+        verifiedInstitutionsByShortname.get(institutionShortName).displayName
+        )
+    );
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionRoleEnum'], undefined));
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleOtherText'], undefined));
+  }
+
+  setInstitutionalRoleOnProfile(institutionalRoleEnum: InstitutionalRole) {
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleEnum'], institutionalRoleEnum));
+    this.setState(fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleOtherText'], undefined));
+  }
+
+  updateVerifiedInstitutionalAffiliation() {
+    const {updatedProfile} = this.state;
```

Comment:
nit: consider further destructuring
```suggestion
const {updatedProfile: {userId, verifiedInstitutionalAffiliation}} = this.state
```

Reply:
you can't use the updatedProfile if you do that. but I can do two lines of destructuring.

Reply:
I don't see updatedProfile being used in this method outside of the property access, that is why I suggested the destructure in that way - is there something I am missing?

Reply:
After a successful save, I change the oldProfile to the updatedProfile so that the check of whether the Save button is enabled works properly.